test/units/TEST-74-AUX-UTILS.run.sh

```diff
@@ -248,13 +248,17 @@ if [[ -e /usr/lib/pam.d/systemd-run0 ]] || [[ -e /etc/pam.d/systemd-run0 ]]; the
     run0 ls /
     assert_eq "$(run0 echo foo)" "foo"
     # Check if we set some expected environment variables
-    for arg in "" "--user=root" "--user=0" "--user=testuser"; do
-        assert_eq "$(run0 ${arg:+"$arg"} bash -c 'echo $SUDO_USER')" "$USER"
-        assert_eq "$(run0 ${arg:+"$arg"} bash -c 'echo $SUDO_UID')" "$(id -u "$USER")"
-        assert_eq "$(run0 ${arg:+"$arg"} bash -c 'echo $SUDO_GID')" "$(id -u "$USER")"
+    for arg in "" "root" "0" "testuser"; do
+        assert_eq "$(run0 ${arg:+"--user=$arg"} bash -c 'echo $SUDO_USER')" "$USER"
+        assert_eq "$(run0 ${arg:+"--user=$arg"} bash -c 'echo $SUDO_UID')" "$(id -u "$USER")"
+        assert_eq "$(run0 ${arg:+"--user=$arg"} bash -c 'echo $SUDO_GID')" "$(id -u "$USER")"
 
         # Validate that we actually went properly through PAM (XDG_SESSION_TYPE is set by pam_systemd)
-        assert_eq "$(run0 ${arg:+"$arg"} bash -c 'echo $XDG_SESSION_TYPE')" "unspecified"
+        assert_eq "$(run0 ${arg:+"--user=$arg"} bash -c 'echo $XDG_SESSION_TYPE')" "unspecified"
+
+        # Validate that $SHELL is set to login shell of target user when cmdline is supplied (not invoking shell)
+        TARGET_LOGIN_SHELL="$(getent passwd "$arg" | cut -d: -f7)"
+        assert_eq "$(run0 ${arg:+"--user=$arg"} printenv SHELL)" "$TARGET_LOGIN_SHELL"
```

Comment:
maybe better to add test case with --machine=, to test the first commit ?

Reply:
We actually can't. To test that run0 must be called without cmdline, which spawns an interactive shell...

Reply:
An appropriate test case will be added in #37071